src/motion_planning/scripts/task_utils.py

```diff
@@ -81,14 +84,21 @@ def transform(origin, destination, odometry=None, pose=None):
     Returns:
     The transformed odometry message
     """
-    if(odometry != None):
-        tfBuffer = tf2_ros.Buffer()
-        listener = tf2_ros.TransformListener(tfBuffer)
-        trans = tfBuffer.lookup_transform(origin, destination, rospy.Time(0))
+    tfBuffer = tf2_ros.Buffer()
+    listener = tf2_ros.TransformListener(tfBuffer)
+    trans = tfBuffer.lookup_transform(origin, destination, rospy.Time(0))
+    if(isinstance(odomORpose, Odometry)):
+        pass
         #TODO: transform odometry    
-    elif(pose != None):
-        tfBuffer = tf2_ros.Buffer()
-        listener = tf2_ros.TransformListener(tfBuffer)
-        trans = tfBuffer.lookup_transform(origin, destination, rospy.Time(0))
-	transformed = tf2_geometry_msgs.do_transform_pose(pose, trans)
+    elif(isinstance(odomORpose, Pose)):
+	    transformed = tf2_geometry_msgs.do_transform_pose(odomORpose, trans)
     return transformed 
+
+def publish_desired_pose_global(pose):
+    TaskState.DESIRED_POSE_GLOBAL_PUBLISHER.publish(pose)
+
+def publish_desired_twist_global(twist):
+    TaskState.DESIRED_TWIST_GLOBAL_PUBLISHER.publish(twist)
```

Comment:
Global variables are a very messy way of dependency injection

Reply:
Again, I think our primary disagreement is about whether the publishers and subscribers are task specific state.